sklearn/cluster/tests/test_birch.py

```diff
@@ -88,9 +96,10 @@ def test_n_clusters(global_random_seed):
         brc4.fit(X)
 
 
-def test_sparse_X(global_random_seed):
+def test_sparse_X(global_random_seed, global_dtype):
     # Test that sparse and dense data give same results
     X, y = make_blobs(n_samples=100, centers=10, random_state=global_random_seed)
+    X = X.astype(global_dtype, copy=False)
     brc = Birch(n_clusters=10)
     brc.fit(X)
```

Comment:
Could you please check the dtype of the `subscluster_centers_` attribute when the model is fit on sparse data?